[CI] SpatialCentroidTests testAggregateIntermediate {TestCase=<geo_point> #2} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3920 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/5rd4jvjtuzepy)
- [elasticsearch-pull-request #29439 / part-3](https://gradle-enterprise.elastic.co/s/sbc6c6t4bnqdk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests.testAggregateIntermediate {TestCase=<geo_point> #2}" -Dtests.seed=A16298C185B81863 -Dtests.locale=ru -Dtests.timezone=Africa/Monrovia -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
yes

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testAggregateIntermediate%20%7BTestCase%3D%3Cgeo_point%3E%20%232%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <[1 1 0 0 0 61 84 1f bc 10 78 13 c0 72 5f 50 ca 3a d7 16 c0]>
     but: was <[1 1 0 0 0 5f 84 1f bc 10 78 13 c0 72 5f 50 ca 3a d7 16 c0]>
```

**Issue Reasons:**
- [main] 2 failures in test testAggregateIntermediate {TestCase=<geo_point> #2} (0.2% fail rate in 917 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -94,12 +94,6 @@ tests:
 - class: org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT
   method: test {yaml=reference/rest-api/watcher/put-watch/line_120}
   issue: https://github.com/elastic/elasticsearch/issues/99517
-- class: org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests
-  method: "testAggregateIntermediate {TestCase=<geo_point> #2}"
-  issue: https://github.com/elastic/elasticsearch/issues/112461
-- class: org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests
-  method: testAggregateIntermediate {TestCase=<geo_point>}
-  issue: https://github.com/elastic/elasticsearch/issues/112463
 - class: org.elasticsearch.xpack.esql.action.ManyShardsIT
   method: testRejection
   issue: https://github.com/elastic/elasticsearch/issues/112406

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialCentroidTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/aggregate/SpatialCentroidTests.java
@@ -22,13 +22,15 @@ import org.elasticsearch.xpack.esql.expression.function.AbstractAggregationTestC
 import org.elasticsearch.xpack.esql.expression.function.FunctionName;
 import org.elasticsearch.xpack.esql.expression.function.MultiRowTestCaseSupplier;
 import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
 
-import java.nio.ByteOrder;
 import java.util.List;
 import java.util.function.Supplier;
 import java.util.stream.Stream;
 
-import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.closeTo;
 
 @FunctionName("st_centroid_agg")
 public class SpatialCentroidTests extends AbstractAggregationTestCase {
@@ -74,16 +76,58 @@ public class SpatialCentroidTests extends AbstractAggregationTestCase {
                 count++;
             }
 
-            var expected = new BytesRef(
-                WellKnownBinary.toWKB(new Point(xSum.value() / count, ySum.value() / count), ByteOrder.LITTLE_ENDIAN)
-            );
+            var expectedX = xSum.value() / count;
+            var expectedY = ySum.value() / count;
 
             return new TestCaseSupplier.TestCase(
                 List.of(fieldTypedData),
                 "SpatialCentroid[field=Attribute[channel=0]]",
                 fieldTypedData.type(),
-                equalTo(expected)
+                centroidMatches(expectedX, expectedY, 1e-14)
             );
         });
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private static Matcher<BytesRef> centroidMatches(double x, double y, double error) {
+        return new TestCentroidMatcher(x, y, error);
+    }
+
+    private static class TestCentroidMatcher extends BaseMatcher<BytesRef> {
+        private final double x;
+        private final double y;
+        private final Matcher<Double> mx;
+        private final Matcher<Double> my;
+
+        private TestCentroidMatcher(double x, double y, double error) {
+            this.x = x;
+            this.y = y;
+            this.mx = closeTo(x, error);
+            this.my = closeTo(y, error);
+        }
+
+        @Override
+        public boolean matches(Object item) {
+            if (item instanceof BytesRef wkb) {
+                var point = (Point) WellKnownBinary.fromWKB(GeometryValidator.NOOP, false, wkb.bytes, wkb.offset, wkb.length);
+                return mx.matches(point.getX()) && my.matches(point.getY());
+            }
+            return false;
+        }
+
+        @Override
+        public void describeMismatch(Object item, Description description) {
+            if (item instanceof BytesRef wkb) {
+                var point = (Point) WellKnownBinary.fromWKB(GeometryValidator.NOOP, false, wkb.bytes, wkb.offset, wkb.length);
+                description.appendText("was ").appendValue(point);
+            } else {
+                description.appendText("was ").appendValue(item);
+            }
+        }
+
+        @Override
+        public void describeTo(Description description) {
+            description.appendValue("    POINT (" + x + " " + y + ")");
+        }
     }
 }

